src/store/checkout-store.js

```diff
@@ -35,6 +36,11 @@ store.registerHandler('NEW_CHECKOUT', data => {
     if (StudentStore.hasOverdueItem(data.studentId)) {
         throw new Error('Student has overdue item');
     }
+    data.itemAddresses.forEach(itemAddress => {
+        if (ItemStore.getItemByAddress(itemAddress).status !== 'AVAILABLE') {
+            throw new Error('An item in the cart is not available for checkout.');
```

Comment:
I think it does, but I haven't tested.

The ItemStore waits on the CheckoutStore, so the error should get thrown before any items are checked out.